R/zchunk_LA100.FAO_downscale_ctry.R

```diff
@@ -232,111 +234,112 @@ module_aglu_LA100.FAO_downscale_ctry <- function(command, ...) {
       mutate(year = as.integer(year)) ->
       FAO_data_ALL_5yr
 
-    # Re-split into separate tables for each element
-    for(i in unique(FAO_data_ALL_5yr$element)) {
-      legacy_name <- paste0("L100.FAO_", i)
-      filter(FAO_data_ALL_5yr, element == i) %>%
-        add_comments("Downscale countries; calculate 5-yr averages") %>%
-        add_legacy_name(legacy_name) %>%
-        add_flags(FLAG_NO_XYEAR, FLAG_LONG_YEAR_FORM) ->
-        df
-      assign(legacy_name, df)
+    ## Re-split into separate tables for each element
+    L100.FAOlist <- split(seq(1,nrow(FAO_data_ALL_5yr)), FAO_data_ALL_5yr$element)
+    names(L100.FAOlist) <- lapply(names(L100.FAOlist), function(x) {paste0("L100.FAO_", x)})
+                                        # change list names to match the legacy
+                                        # names
+    fixup <- function(irows, legacy.name) {
+        FAO_data_ALL_5yr[irows,] %>%
+          add_comments("Downscale countries; calculate 5-yr averages") %>%
+          add_legacy_name(legacy.name) %>%
+          add_flags(FLAG_NO_XYEAR, FLAG_LONG_YEAR_FORM)
     }
+    L100.FAOlist <- Map(fixup, L100.FAOlist, names(L100.FAOlist))
```

Comment:
Map can take more than one list as an argument.  In this case it takes both the elements of the list and their names.